cpp/oneapi/dal/test/engine/thread_communicator.hpp

```diff
@@ -358,11 +415,11 @@ class thread_communicator_impl
             : base_t(queue),
               ctx_(thread_count),
               barrier_(ctx_),
-              bcast_(ctx_),
+              bcast_(ctx_, queue),
               gather_(ctx_),
-              allgatherv_(ctx_),
-              sendrecv_replace_(ctx_),
-              allreduce_(ctx_),
+              allgatherv_(ctx_, queue),
+              sendrecv_replace_(ctx_, queue),
```

Comment:
as a potential approach, can we have 2 comm function definitions:
sendrecv_replace(ctx)
sendrecv_replace(ctx, queue)
for dispatching between usm and host memory